src/nouislider.js

```diff
@@ -1100,6 +1111,53 @@
             return addNodeTo(handle.firstChild, options.cssClasses.tooltip);
         }
 
+        // create or update joint tooltips and hide or display related tooltips
+        function updateJointTooltips() {
+            for (var index in scope_OverlappingTooltips) {
+                var jointTooltip = scope_JointTooltips[index];
+                var overlapInfo = scope_OverlappingTooltips[index];
+
+                var firstHandleIndex = overlapInfo[0];
+                var secondHandleIndex = overlapInfo[1];
+
+                var firstTooltip = scope_Tooltips[firstHandleIndex];
+                var secondTooltip = scope_Tooltips[secondHandleIndex];
+
+                // if overlapping tooltips info has 3 values, two overlapping tooltips are separated and joint tooltip
+                // can be removed and original tooltips can be visible again
+                //
+                // else if overlapping tooltip is not existing then create it and gide related tooltips
+                //
+                // else update position of overlapping tooltip
+                if (overlapInfo.length === 3) {
+                    firstTooltip.className = options.cssClasses.tooltip;
+                    secondTooltip.className = options.cssClasses.tooltip;
+                    jointTooltip.remove();
+                    delete scope_JointTooltips[index];
+                    delete scope_OverlappingTooltips[index];
+                } else if (!jointTooltip) {
+                    firstTooltip.className += " noUi-hidden-tooltip";
+                    secondTooltip.className += " noUi-hidden-tooltip";
+                    var newTooltip = addNodeTo(scope_Handles[index].firstChild, options.cssClasses.tooltip);
+                    scope_JointTooltips[index] = newTooltip;
+
+                    var location = -(scope_Locations[secondHandleIndex] - scope_Locations[firstHandleIndex]) / 2;
+                    newTooltip.style.marginLeft = (location * scope_Target.offsetWidth) / 100 + "px";
```

Comment:
Both changing `marginLeft` and accessing `offsetWidth` here can cause expensive redraws. Since this method is in the hot path of user interaction, that's not desirable. The margin offsets could be replaced by CSS transforms.